README.md

```diff
@@ -1,208 +1,262 @@
 
-### Step 6 - Setup Mendix app
-Open up your application in Mendix and login as an Admin, so that you can see the menu option google admin.
-Enter the project ID into the sender ID field and the API key into the API field. 
-<img src="assets/images/gcm-step6.PNG"/>
+<img src="assets/images/testing/send message.JPG"/>
 
-Once entered tick the checkbox enabled and press restart. From now on your application will always start the gcm push notification system for you.
+## APIs
 
-For more information on setting up your google API then please refer to this article: (Google API Setup) [http://developer.android.com/google/gcm/gs.html]
+This is a list of microflows that can be called from within your application. They are located in the `_USE ME/Microflows` folder.
 
+|Microflow name         |Description                                                    |
+|-----------------------|---------------------------------------------------------------|
+|MessageFactory         |A microflow to create a `Message` object                       |
+|SendMessageImmediately |A microflow to send a message to GCM/APNs immediately          |
+|SendMessageQueued      |A microflow to send a message to GCM/APNs using a queue        |
+|SendMessagesImmediately|A microflow to send multiple messages to GCM/APNs immediately  |
+|SendMessagesQueued     |A microflow to send multiple messages to GCM/APNs using a queue|
 
-## Setting up Windows
-Windows requires no additional configuration for push notifications to work. Simply load the application onto a Windows 8 Phone and login to the application. The user's Windows Phone credentials should appear within the device menu.
-The notifications for Windows work using a web service, the widget within your application registers the user's unique URL. With the URL the Mendix application can pass a message to this URL.  
+Note: for sending message(s) queue to work properly, you need to ensure that the scheduled event `PushQueue` is active. In a sandbox/free app this does not require any extra step, but in a TAP environments you will need to explicitly set the scheduled event to be activated.
 
+## Setting up Apple Push Notification Server
 
-## Installing a Windows App
-In order to test and publish your Windows 8 applications you will need a Windows developer account that can be obtained from: 
+In order to proceed you need an Apple developer license and a device running Mac OS X.
 
-[Windows Dev Center](https://dev.windows.com/en-us) 
+We assume that you already have the app signing key with provisioning profile and can freely build and install your mobile app (if not, please refer to [this howto] (https://world.mendix.com/display/howto50/Publishing+a+Mendix+Hybrid+Mobile+App+in+Mobile+App+Stores#PublishingaMendixHybridMobileAppinMobileAppStores-3.SettingupAppSigningKeys)). Take into account that your App ID should use Explicit App ID and have Push Notification turned on so you could receive push notifications with your app.
```

Comment:
how-to

and let's link to the MX 6 version:
https://world.mendix.com/display/howto6/Publishing+a+Mendix+Hybrid+Mobile+App+in+Mobile+App+Stores#PublishingaMendixHybridMobileAppinMobileAppStores-3.SettingupAppSigningKeys

- I think I'm missing a 'the or an' in front of Explicit App ID, not sure which of the two.

- Push Notifications should be plural I think.

- 'so you can' IMO

- Not entirely sure we explain sufficiently how to enable push notifications here?